aion/dialyzer.ignore_warnings

```diff
@@ -14,3 +14,8 @@ web/channels/room_channel.ex:14: The inferred return type of join/3 ({'ok',#{'__
 lib/aion/room_channel/room.ex:51: Invalid type specification for function 'Elixir.Aion.RoomChannel.Room':change_question/2. The success typing is (atom() | #{'__struct__':=atom(), atom()=>_},binary()) -> #{'__struct__':=atom(), atom()=>_}
 lib/aion/room_channel/room.ex:79: Invalid type specification for function 'Elixir.Aion.RoomChannel.Room':get_new_question_with_answers/1. The success typing is (binary()) -> #{'answers':=[map()], 'question':=#{'__meta__':=map(), '__struct__':='Elixir.Aion.Question', 'category':=map(), 'category_id':='nil', 'content':=_, 'id':=integer(), 'image_name':=_, 'inserted_at':='nil', 'updated_at':='nil'}}
 web/models/question.ex:39: Invalid type specification for function 'Elixir.Aion.Question':get_random_question/1. The success typing is (binary()) -> #{'__meta__':=#{'__struct__':='Elixir.Ecto.Schema.Metadata', 'context':='nil', 'source':={'nil',<<_:72>>}, 'state':='built'}, '__struct__':='Elixir.Aion.Question', 'category':=#{'__cardinality__':='one', '__field__':='category', '__owner__':='Elixir.Aion.Question', '__struct__':='Elixir.Ecto.Association.NotLoaded'}, 'category_id':='nil', 'content':=_, 'id':=_, 'image_name':=_, 'inserted_at':='nil', 'updated_at':='nil'}
+web/channels/room_channel.ex:12: The return type {'ok',#{'__struct__':='Elixir.Phoenix.Socket', 'assigns':=_, 'channel':=_, 'channel_pid':=_, 'endpoint':=_, 'handler':=_, 'id':=_, 'joined':=_, 'pubsub_server':=_, 'ref':=_, 'serializer':=_, 'topic':=_, 'transport':=_, 'transport_name':=_, 'transport_pid':=_}} in the specification of join/3 is not a subtype of {'error',map()} | {'ok',#{'__struct__':='Elixir.Phoenix.Socket', 'assigns':=#{}, 'channel':=atom(), 'channel_pid':=pid(), 'endpoint':=atom(), 'handler':=atom(), 'id':='nil', 'joined':=boolean(), 'pubsub_server':=atom(), 'ref':=_, 'serializer':=atom(), 'topic':=binary(), 'transport':=atom(), 'transport_name':=atom(), 'transport_pid':=pid()}} | {'ok',map(),#{'__struct__':='Elixir.Phoenix.Socket', 'assigns':=#{}, 'channel':=atom(), 'channel_pid':=pid(), 'endpoint':=atom(), 'handler':=atom(), 'id':='nil', 'joined':=boolean(), 'pubsub_server':=atom(), 'ref':=_, 'serializer':=atom(), 'topic':=binary(), 'transport':=atom(), 'transport_name':=atom(), 'transport_pid':=pid()}}, which is the expected return type for the callback of the 'Elixir.Phoenix.Channel' behaviour
```

Comment:
same here ;)

Reply:
Did you check the origin of this error? It actually is heavily connected to the introduced logic :smile:

Reply:
Sorry about that. I didn't dive into it :P